src/Microsoft.AspNetCore.Sockets.Client/LongPollingTransport.cs

```diff
@@ -171,43 +176,76 @@ private IList<Message> ParsePayload(byte[] payload)
         private async Task SendMessages(Uri sendUrl, CancellationToken cancellationToken)
         {
             _logger.LogInformation("Starting the send loop");
-
-            TaskCompletionSource<object> sendTcs = null;
+            IList<SendMessage> messages = null;
             try
             {
                 while (await _application.Input.WaitToReadAsync(cancellationToken))
                 {
+                    // Grab as many messages as we can from the channel
+                    messages = new List<SendMessage>();
                     while (!cancellationToken.IsCancellationRequested && _application.Input.TryRead(out SendMessage message))
                     {
-                        sendTcs = message.SendResult;
+                        messages.Add(message);
+                    }
+
+                    if (messages.Count > 0)
+                    {
+                        _logger.LogDebug("Sending {0} message(s) to the server using url: {1}", messages.Count, sendUrl);
+
+                        // Send them in a single post
                         var request = new HttpRequestMessage(HttpMethod.Post, sendUrl);
                         request.Headers.UserAgent.Add(DefaultUserAgentHeader);
 
-                        if (message.Payload != null && message.Payload.Length > 0)
-                        {
-                            request.Content = new ByteArrayContent(message.Payload);
-                        }
+                        // TODO: We can probably use a pipeline here or some kind of pooled memory.
+                        // But where do we get the pool from? ArrayBufferPool.Instance?
+                        var memoryStream = new MemoryStream();
 
-                        _logger.LogDebug("Sending a message to the server using url: '{0}'. Message type {1}", sendUrl, message.Type);
+                        // Write the messages to the stream
+                        var pipe = memoryStream.AsPipelineWriter();
+                        var output = new PipelineTextOutput(pipe, TextEncoder.Utf8); // We don't need the Encoder, but it's harmless to set.
```

Comment:
It's only needed because the only implementation of `IOutput` for Pipelines also implements `ITextOutput` which has an Encoder property, which the constructor sets. I set it to `Utf8` because that's the right format for any text we want to output and I didn't want to risk a null-ref (though to be fair, it should work with `null`)